### ISSUE TITLE
End-to-end tests does not work on macOS

### ISSUE THREAD

**Issue by samuel-utbult-oborgen** [HoudiniGraphql/houdini#695]
### Describe the bug

The end-to-end tests cannot be run on macOS 12.6. Going to any page in the website causes a 500 Internal Server Error to be thrown and the following is printed in the terminal:

```
[web] fetch failed
[web] TypeError: fetch failed
[web]     at fetch (/path/to/houdini/node_modules/.pnpm/undici@5.11.0/node_modules/undici/index.js:105:13)
[web]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[web]     at async fetch (file:///path/to/houdini/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.510_svelte@3.52.0+vite@3.1.4/node_modules/@sveltejs/kit/src/runtime/server/fetch.js:60:21)
[web]     at async Object.fetch (file:///path/to/houdini/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.510_svelte@3.52.0+vite@3.1.4/node_modules/@sveltejs/kit/src/runtime/server/fetch.js:22:10)
[web]     at async Object.fetch (file:///path/to/houdini/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.510_svelte@3.52.0+vite@3.1.4/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:102:21)
[web]     at async fetch (/$houdini/runtime/lib/network.js:50:26)
[web]     at async HoudiniClient.fetchQuery [as fetchFn] (/path/to/houdini/e2e/sveltekit/src/lib/graphql/houdiniClient.ts:16:19)
[web]     at async HoudiniClient.sendRequest (/$houdini/runtime/lib/network.js:47:20)
[web]     at async Module.fetchQuery (/$houdini/runtime/lib/network.js:154:18)
[web]     at async LayoutSessionStore.fetchAndCache (/$houdini/plugins/houdini-svelte/runtime/stores/query.js:121:21)
```

Additionally, most tests time out if `pnpm test` is run, most probably because of this bug. I have tested to set up the end-to-end tests on Windows 10 and Ubuntu 22.04 and the environment is working just fine on those platforms (with the exception of #656 being present on Windows 10).

### Severity

serious, but I can work around it

### Steps to Reproduce the Bug

1. Use macOS 12.6.
2. Set up the project in accordance with https://www.houdinigraphql.com/guides/contributing#local-development
3. Go to localhost:5173
4. See the bug.

**Comment by AlecAivazis** [comment]
Huh, any ideas why? I don't have an OSX machine I can test this on so if you have the time to sort through this it would be very appreciated. I can offer any guidance you need here or on discord

**Comment by samuel-utbult-oborgen** [comment]
It looks like in the stack trace that the library undici is the reason why it doesn't work. However, it seems to be used through SvelteKit, and I haven't seen this error in my regular SvelteKit project that usually runs on macOS. I am currently quite busy right now and it looks like I can get around this problem for the moment but I will try to see what I can do.

**Comment by samuel-utbult-oborgen** [comment]
An observation I just made is that the same error message is displayed on Ubuntu when an error is introduced in the Yoga server that causes the server to crash. This could possibly mean that something could be wrong in the communication between SvelteKit and Yoga causing SvelteKit to not find Yoga on macOS.

**Comment by samuel-utbult-oborgen** [comment]
I figured out a solution. Add `hostname: "::",` to `e2e/_api/server.mjs` at line 9. The problem is caused by NodeJS (version 19.0.0) for some reason refusing connections from other NodeJS instances when running on macOS, no hostname is specified and a request is made with the host `localhost`. Using `localhost` with any other client (I have tested a web browser, Postman and curl) works just fine. The same behavior is displayed when creating a barebones NodeJS server. Notice that `::` means the server will be open to connections from any other computer on the network (as also is the case right now). Setting the hostname to `localhost` seems to work as well as long as you would be fine with the API only being reachable from the computer the API is running on, which is something other users might complain about if they have certain requirements. I can create a PR with the proposed fix.
